Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import XCGLogger
+import Deferred
+
+/// Small enum to help strength parameter requirements for potential bookmark highlights query below.
+private enum BookmarkSource {
+    case Mirror
+    case Local
+
+    var modifiedColumn: String  {
+        switch self {
+        case Mirror: return "server_modified"
+        case Local: return "local_modified"
+        }
+    }
+
+    var tableName: String {
+        switch self {
+        case Mirror: return TableBookmarksMirror
+        case Local: return TableBookmarksLocal
+        }
+    }
+}
+
+extension SQLiteHistory: HistoryRecommendations {
+    public func getHighlights() -> Deferred<Maybe<Cursor<Site>>> {
+        let limit = 19
+        let microsecondsPerMinute: Timestamp = 60_000_000 // 1000 * 1000 * 60
+        let thirtyMinutesAgo = NSDate.nowMicroseconds() - 30 * microsecondsPerMinute
+        let threeDaysAgo = NSDate.nowMicroseconds() - (60 * microsecondsPerMinute) * 24 * 3
+        let bookmarkLimit = 1
+        let historyLimit = limit - bookmarkLimit
+
+        let recentHistory = [
+            "SELECT *",
+            "FROM (",
+            "   SELECT \(TableHistory).id AS historyID, url, title, guid, sum(1) AS visitCount, max(\(TableVisits).date) AS visitDate",
+            "   FROM \(TableHistory)",
+            "   LEFT JOIN \(TableVisits) ON \(TableVisits).siteID = \(TableHistory).id",
+            "   WHERE title NOT NULL AND title != '' AND is_deleted = 0",
+            "   GROUP BY url",
+            "   ORDER BY visitDate DESC",
+            "   LIMIT \(historyLimit)",
+            ")",
+            "WHERE visitCount <= 3 AND visitDate < \(thirtyMinutesAgo)",
+        ].joinWithSeparator(" ")
+
+        let bookmarkHighlights = [
+            "SELECT historyID, url, title, guid, visitCount, visitDate",
+            "FROM (",
+            "   SELECT *",
```

Comment:
> Note that there's already a view that overlays Local on top of Mirror, so use that

I'd like to use the view but it has no notion of 'modified' time. I suppose I could add it to the projection but then it would combine `server_modified` and `local_modified` under the same column. Given that the clocks aren't exact, I'm not sure if that would be safe or not.

Thoughts?